### PR TITLE
feat: setup internationalization

### PR DESCRIPTION
**Pull request by VadimTarasenko**

### CODE REVIEW

File: src/app/[lang]/auth/sign-up/page.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+// utils
+import { getDictionary } from '@/modules/internationalization/utils/dictionary.utils';
+// types
+import type { PageParams } from '@/modules/core/types/next.types';
+
+export default async function Page({ params }: PageParams) {
+  const dictionary = await getDictionary(params.lang);
+
+  return (
+    <IntlProvider locale={params.lang} dictionary={dictionary}>
```
[the remark this answers is not in the document]
Comment (VadimTarasenko) [reply src/app/[lang]/auth/sign-up/page.tsx:13]: Then we kills ssr forever. IntlProvider is a client component and if we add it to layout, all pages should have 'use client' directive. I suggest to write this provider separately on each page;

Next way:
```tsx
export const MyPage = async ({ params }) => {
  const dictionary = await getDictionary(params.lang);

  return (
    <main>
      <div>{dictionary.myPage.title}</div> // rendered on server
      <IntlProvider>
         <ClientComponent />
      </IntlProvider>
    </main>
  )
}
```

File: src/app/[lang]/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import '@/styles/globals.scss';
+// constants
+import { i18nConfig } from '@/modules/internationalization/constants/i18n.constants';
+// providers
+import { TrpcProvider } from '@/utils/trpc-provider';
+// types
+import type { ChildrenProp } from '@/modules/core/types/react.types';
+import type { PageParams } from '@/modules/core/types/next.types';
+
+export default function RootLayout({
```
[the remark this answers is not in the document]
Comment (VadimTarasenko) [reply src/app/[lang]/layout.tsx:10]: Now `src/app/[lang]` is our root for all pages.

Screenshot 2023-08-23 at 11 37 36

https://nextjs.org/docs/app/building-your-application/routing/internationalization

[the remark this answers is not in the document]
Comment (VadimTarasenko) [reply src/app/[lang]/layout.tsx:10]: It's also possible to move layout to app. But I think it's good if our "index" page and "index" layout will be placed together. What do you think?

[the remark this answers is not in the document]
Comment (IhorBalan) [reply src/app/[lang]/layout.tsx:10]: I don't think that's a good idea, as the `[lang]` is dynamic path. While it seems logical that we will place all the page into the subdirectories, it acctually doesn't change the fact that our app lives in tha `app` folder, so all configuration that applies to whole app should be placed there.

[the remark this answers is not in the document]
Comment (IhorBalan) [reply src/app/[lang]/layout.tsx:10]: I see that they had this comment in the doc, but personally I'm not sure if it's a good idea.

**Pull request merged**